src/Sets/SparsePolynomialZonotope.jl

```diff
@@ -0,0 +1,296 @@
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Notes
+
+This is equivalent to the number of monomials in the polynomial representation of `P`.
+"""
+ngens(P::SPZ) = ndependentgens(P) + nindependentgens(P)
+
+"""
+    nparams(P::SparsePolynomialZonotope)
+
+Return the number of dependent parameters in the polynomial representation of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The number of dependent parameters in the polynomial representation of P.
+
+### Notes
+
+This corresponds to the number rows in the exponent matrix ``E``.
+"""
+nparams(P::SPZ) = size(P.E, 1)
+
+"""
+    order(P::SparsePolynomialZonotope)
+
+Return the order of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The order of `P`, defined as the quotient between the number of generators and the ambient dimension.
+"""
+order(P::SPZ) = ngens(P) // dim(P)
+
+"""
+    center(P::SparsePolynomialZonotope)
+
+Return the center of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The center of `P`.
+"""
+center(P::SPZ) = P.c
+
+"""
+    dependent_genmat(P::SparsePolynomialZonotope)
+
+Return the matrix of dependent generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+dependent_genmat(P::SPZ) = P.G
+
+"""
+    independent_genmat(P::SparsePolynomialZonotope)
+
+Return the matrix of independent generators of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+"""
+independent_genmat(P::SPZ) = P.GI
+
+"""
+    expmat(P::SparsePolynomialZonotope)
+
+Return the matrix of exponents of the sparse polynomial zonotope.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The matrix of exponents, where each column is a multidegree.
+
+### Notes
+
+In the exponent matrix, each row corresponds to a parameter (``\alpha_k`` in the definition) and each column to a monomial.
+"""
+expmat(P::SPZ) = P.E
+
+"""
+    indexvector(P::SparsePolynomialZonotope)
+
+Return the index vector of `P`.
+
+### Input
+
+- `P` -- sparse polynomial zonotope
+
+### Notes
+
+The index vector is a vector of positive integers identifing the dependent parameters of `P`.
+"""
+indexvector(P::SPZ) = P.idx
+
+"""
+    linear_map(M::AbstractMatrix, P::SparsePolynomialZonotope)
+
+Apply the linear map `M` to the sparse polynomial zonotope `P`.
+
+### Input
+
+- `M` -- square matrix with size(M) == dim(P)
+- `P` -- sparse polynomial zonotope
+
+### Output
+
+The set resulting from applying the linear map `M` to `P`.
+"""
+function linear_map(M::AbstractMatrix, P::SPZ)
+    return SparsePolynomialZonotope(M * center(P),
+                                    M * dependent_genmat(P),
+                                    M * independent_genmat(P),
+                                    expmat(P),
+                                    indexvector(P)
+                                    )
+end
+
+# function quadratic_map(Q::Vector{MT}, S::SparsePolynomialZonotope) where {N, MT<:AbstractMatrix{N}}
```

Comment:
There is some commented code at the end of the file. Is this intended?

Reply:
I created this file from `SparsePolynomialZonotopes.jl` and then adapted. `quadratic_map` is not implemented for SPZ, but I just commented this out instead of deleting it to leave it as a starting point for the quadratic map, I can also delete it and just copy paste the base code when the time comes

Reply:
Okay, just wanted to make sure you did not forget something.